### PR TITLE
feat(api): new domain for health directorate on service portal

### PR DESCRIPTION
**Pull request by disaerna**
# Api - Health Directorate

## What

Domain layer for health part from directorate of health in service portal

## Why

Part of moving heilsuvera to mínar síður.

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
	- Introduced a new domain for managing health-related data, including organ donation and vaccinations.
	- Added a resolver and service for handling GraphQL queries and mutations related to donor status and vaccinations.
	- Created essential configuration files for linting, testing, and TypeScript settings specific to the new domain.
  
- **Documentation**
	- Added a README file with instructions on the library's purpose and usage.

### CODE REVIEW

File: libs/api/domains/health-directorate/src/lib/health-directorate.resolver.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  DonorStatus,
+  DonorStatusInput,
+} from './models/organ-donation.model'
+import type { Locale } from '@island.is/shared/types'
+import { Vaccinations } from './models/vaccinations.model'
+import { HealthDirectorateService } from './health-directorate.service'
+
+@UseGuards(IdsUserGuard, ScopesGuard)
+@Scopes(ApiScope.internal)
+@Resolver()
```
Comment (thordurhhh) [inline libs/api/domains/health-directorate/src/lib/health-directorate.resolver.ts:24]: Maybe we should add audit here?

```
@Audit({ namespace: '@island.is/api/health-directorate' })

...


  @Audit()
```

Comment (disaerna) [reply libs/api/domains/health-directorate/src/lib/health-directorate.resolver.ts:24]: Yebb! 💯 I just forgot about it existing 🙈

**Review by thordurhhh — APPROVED** [review]
Nice! 🙌

**Review by svanaeinars — APPROVED** [review]
LGTM

**Review by eirikurn — APPROVED** [review]

**Pull request merged**